MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,386 @@
+            /// </summary>
+            /// <returns>Random integer of type Int16 (short)</returns>
+            public short NextShort()
+            {
+                // Shift to upper 15 bits for positive values only, then cast
+                return (short)(PcgXshRS() >> 17);
+            }
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int16 (short) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Exclusive maximum value.</param>
+            /// <returns>Random integer of type Int16 (short)</returns>
+            public short NextShort(short max)
+            {
+                return NextShort(0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int16 (short) that is within a specified range.
+            /// </summary>
+            /// <param name="min">Inclusive minimum value.</param>
+            /// <param name="max">Exclusive maximum value.</param>
+            /// <returns>Random integer of type Int16 (short)</returns>
+            public short NextShort(short min, short max)
+            {
+                // Cast to int, in case the total range is bigger than short.MaxValue
+                int range = (int)max - (int)min;
+
+                return (short)((PcgXshRS() % range) + min);
+            }
+
+            #endregion
+
+            #region Int32 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int32 (int).
+            /// </summary>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next()
+            {
+                // Shift to remove sign bit, then cast
+                return (int)(PcgXshRS() >> 1);
+            }
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int32 (int) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Exclusice maximum value.</param>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next(int max)
+            {
+                return Next(0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int32 (int) that is within a specified range.
+            /// </summary>
+            /// <param name="min">Inclusive minimum value.</param>
+            /// <param name="max">Exclusive maximum value.</param>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next(int min, int max)
+            {
+                // Cast to uint in case the total range is greater than int.MaxValue
+                // Force overflow on negative numbers, so they remain less than positive ones
+                uint _min = (uint)min + 0x80000000;
+                uint _max = (uint)max + 0x80000000;
```

Comment:
What if the user passes an inverted min/max values?  Does this still produce a reasonable random?  Should we add a debug only assert to catch errors like this?